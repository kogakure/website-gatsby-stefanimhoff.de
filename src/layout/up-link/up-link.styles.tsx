import styled from 'styled-components';

import { transitionDuration } from '../../layout/tokens';
import { ArrowUp } from '../../components/icons';

const Button = styled.button`
  align-items: center;
  background: transparent;
  border: 0;
  cursor: pointer;
  display: flex;
  height: 40px;
  justify-content: center;
  width: 40px;
`;

const Arrow = styled(ArrowUp)`
  fill: var(--colorForeground);
  font-size: 20px;
`;

const ArrowLink = styled.a`
  transition-duration: ${transitionDuration[5]};
  transition-property: transform;
  transition-timing-function: ease-in-out;

  &:hover,
  &:focus {
    transform: translate3D(0, -0.25rem, 0);
  }
`;

export const Styled = {
  Arrow,
  ArrowLink,
  Button,
};
