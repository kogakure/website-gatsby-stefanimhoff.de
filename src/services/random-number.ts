export const getRandomNumber = (max: number) =>
  String(Math.floor(Math.random() * Math.floor(max)));
