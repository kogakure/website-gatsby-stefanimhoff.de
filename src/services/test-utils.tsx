import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
