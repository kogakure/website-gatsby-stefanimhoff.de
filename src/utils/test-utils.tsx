import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
