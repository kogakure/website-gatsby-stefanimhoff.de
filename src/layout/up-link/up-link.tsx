import * as React from 'react';

import { Styled } from './up-link.styles';

const scrollToTop = (e: React.MouseEvent) => {
  e.preventDefault();
  window.scroll({
    top: 0,
    left: 0,
    behavior: 'smooth',
  });
};

export const UpLink: React.FC = ({ ...props }) => (
  <Styled.ArrowLink href="#top" onClick={scrollToTop} {...props}>
    <Styled.Button type="button" aria-label="Back to top">
      <Styled.Arrow aria-hidden="true" />
    </Styled.Button>
  </Styled.ArrowLink>
);
