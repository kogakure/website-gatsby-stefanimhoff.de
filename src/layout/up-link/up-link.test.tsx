import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { UpLink } from './up-link';

describe('UpLink', () => {
  test('renders correctly', () => {
    const { container } = render(<UpLink />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('is accessible', async () => {
    const { container } = render(<UpLink />);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
