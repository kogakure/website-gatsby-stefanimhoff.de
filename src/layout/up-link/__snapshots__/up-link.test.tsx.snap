// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpLink renders correctly 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
}

.c2 {
  fill: var(--colorForeground);
  font-size: 1.2rem;
}

.c0 {
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
}

.c0:hover,
.c0:focus {
  -webkit-transform: translate3D(0,-0.25rem,0);
  -ms-transform: translate3D(0,-0.25rem,0);
  transform: translate3D(0,-0.25rem,0);
}

<a
  class="c0"
  href="#top"
>
  <button
    aria-label="Back to top"
    class="c1"
    type="button"
  >
    <svg
      class="c2"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M13 7.828V20h-2V7.828l-5.364 5.364-1.414-1.414L12 4l7.778 7.778-1.414 1.414L13 7.828z"
      />
    </svg>
  </button>
</a>
`;
